src/test/java/teammates/test/driver/TestProperties.java

```diff
@@ -165,20 +166,28 @@ private static String extractVersionNumber(String inputString) {
 
     /**
      * Verifies that the test properties specified in test.properties file allows for HTML
-     * regeneration via God mode to work smoothly (i.e all test HTML files are correctly regenerated,
+     * regeneration via GodMode to work smoothly (i.e all test HTML files are correctly regenerated,
      * strings that need to be replaced with placeholders are correctly replaced, and
      * strings that are not supposed to be replaced with placeholders are not replaced).
      */
     public static void verifyReadyForGodMode() {
+        if (IS_GODMODE_ENABLED && isCiEnvironment() && !isGodModeTest()) {
+            fail("GodMode should only be run locally, not in a CI environment. Please revert the change "
+                    + "to the test properties template file that enabled GodMode in CI.");
+        }
         if (!isDevServer()) {
-            fail("God mode regeneration works only in dev server.");
+            fail("GodMode regeneration works only in dev server.");
         }
         if (isStudentMotdUrlEmpty()) {
             fail("Student MOTD URL defined in app.student.motd.url in build.properties "
                     + "must not be empty. It is advised to use test-student-motd.html to test it.");
         }
     }
 
+    private static boolean isGodModeTest() {
+        return System.getProperty("test.godmode") != null;
```

Comment:
I don't think this property helps. System properties is, well, system-wide, and thus there's still bound to be that short period where CI might run with God mode on.